### PR TITLE
Logging fixes

### LINKED ISSUE #1454: td-agent may not restart after node crash
td-agent crashes with
```
2019-04-07 15:55:50 +0000 [error]: fluent/log.rb:362:call: config error file="/td-agent/config/fluent.conf" error_class=Fluent::ConfigError error="Unexpected error: failed to read data from plugin storage file: '/var/log/journald.pos/worker0/storage.json'"
```

/var/log/journald.pos/worker0/storage.json is 0 bytes long

root cause is probably that fluentd doesn't call fdatasync

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

fixes #1454 

* collect core dump dmesgs to artifacts, if they exist
* revamp hack/ssh.sh and hack/scp.sh
* bump image version
* convert utf-8 messages to ascii (otherwise we lose log lines @charlesakalugwu @mjudeikis)
* minimise buffering on mdsd plugin, stop writing to disk, make blocking
* fix go get imagebuilder
* clean up multiple directories in /var/crash
* workaround: remove empty fluentd state files if needed after hard crash (see #1454)
* pull audit files out of journal, log quantity coupled with limited os disk io makes journal hard to use for anything else (@thekad fyi)

@thekad @charlesakalugwu I've given up with logging crash dmesg to journald for now - seems impossible to configure mdsd to be sure it gets into Geneva
@charlesakalugwu ditto read_from_head :-(

### CODE REVIEW

File: pkg/startup/v5/data/master/etc/origin/master/master-config.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ admissionConfig:
         kind: ImagePolicyConfig
 auditConfig:
   enabled: true
+  auditFilePath: /var/log/openshift-audit/openshift-audit.log
```
Comment (mjudeikis) [inline pkg/startup/v5/data/master/etc/origin/master/master-config.yaml]: +1

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ data:
         @type local
         path /var/log/journald.pos
       </storage>
+      read_from_head true
```
Comment (charlesakalugwu) [inline pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: we might also want to add `read_from_head` [here](https://github.com/openshift/openshift-azure/blob/5951f298a5ce20d01b9f06eabb99b7b29a8dfc31/pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml#L19)?

**Comment by charlesakalugwu** [comment]
@jim-minter generally looks good to me. I'll now try to create a fakerp cluster with these changes to see if that fixes #1395

**Comment by charlesakalugwu** [comment]
@jim-minter machine boot logs are still not being forwarded. Check the cluster `charles-dev-tdagent`

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml (hunk 2 above)
Comment (jim-minter) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: I read the docs (and the code) and believe that we shouldn't - I think that if you specify multiple files fluentd is supposed to "do the right thing" - am going to test manually

**Comment by charlesakalugwu** [comment]
@jim-minter 

seeing lots of errors in the td-agent logs

```
2019-04-09 11:05:30 +0000 [info]: starting fluentd worker pid=9922 ppid=9903 worker=0
2019-04-09 11:05:30 +0000 [info]: fluentd worker is now running worker=0
2019-04-09T11:05:33.2994760Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:33.4498740Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:33.7763000Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:34.6795190Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:36.2483700Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:39.5053620Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:45.0168460Z: error SocketClient.cc:137 Connect() SocketException: SocketClient connect(): No such file or directory
2019-04-09T11:05:54.9351100Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_INVALID_SOURCE, tag '1'
2019-04-09T11:05:54.9410070Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:05:55.0569220Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:05:55.0701450Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09 11:05:55 +0000 [warn]: buffer flush took longer time than slow_flush_log_threshold: elapsed_time=22.167242299999998 slow_flush_log_threshold=20.0 plugin_id="object:3f8f32ce5a80"
2019-04-09T11:05:56.4567030Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_INVALID_SOURCE, tag '1897'
2019-04-09T11:06:33.3403750Z: warn DataReader.cc:230 tag '19628' is not found in backup cache
2019-04-09T11:06:33.3404120Z: warn DataReader.cc:230 tag '19627' is not found in backup cache
2019-04-09T11:06:49.2435480Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:50.0462070Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:50.0549350Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:51.1518800Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:51.1553630Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:51.2976450Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
2019-04-09T11:06:51.8707570Z: error DataReader.cc:224 unexpected mdsd ack status: ACK_DECODE_ERROR, tag '0'
```

**Comment by jim-minter** [comment]
/hold cancel

@asalkeld @mjudeikis would like to merge this asap because hopefully it'll catch crashes if they're happening

**Comment by asalkeld** [comment]
flake https://github.com/openshift/openshift-azure/issues/1410
/retest

**Comment by charlesakalugwu** [comment]
@jim-minter needs rebase

**Comment by mjudeikis** [comment]
rebased
/lgtm

**Pull request merged**